File: Server_Spring/Elderly_Care/src/main/java/kr/spring/care/user/entity/Caregiver.java
package kr.spring.care.user.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Caregiver {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long caregiverId;
    private String certification;
    private String specialization;
    private String experience;
    private int experienceYears;
    private String availableHours;

    @JsonIgnore
    @OneToOne(cascade = CascadeType.REMOVE)
    @JoinColumn(name = "user_id")
    private User user;
}

File: Client_Android/app/src/main/java/com/example/elderlycare/matching/model/Caregiver.kt
package com.example.elderlycare.matching.model

data class Caregiver(
    val name: String,
    val country: String,
    val experience: String,
    val experienceYears: Int,
    val certification: String,
    val availableHours: String,
    val image: String,
    val gender: String,
    val caregiverId: Int,
)